libraries/chain/protocol/transaction.cpp

```diff
@@ -338,67 +336,65 @@ const flat_set<public_key_type>& signed_transaction::get_signature_keys( const c
+   sign_state s(signature_keys, get_active, get_owner, allow_non_immediate_owner, max_recursion_depth, available_keys);
 
-   for( const auto& auth : other )
+   for (const auto& auth : other)
       s.check_authority(&auth);
-   for( auto& owner : required_owner )
-      s.check_authority( get_owner( owner ) );
-   for( auto& active : required_active )
-      s.check_authority( active ) || s.check_authority( get_owner( active ) );
+   for (auto& owner : required_owner)
+      s.check_authority(get_owner(owner));
+   for (auto& active : required_active)
+      s.check_authority(active) || s.check_authority(get_owner(active));
 
    s.remove_unused_signatures();
 
    set<public_key_type> result;
 
-   for( auto& provided_sig : s.provided_signatures )
-      if( available_keys.find( provided_sig.first ) != available_keys.end()
-            && signature_keys.find( provided_sig.first ) == signature_keys.end() )
-         result.insert( provided_sig.first );
+   for (auto& provided_sig : s.provided_signatures)
+      if (available_keys.find(provided_sig.first) != available_keys.end()
+            && signature_keys.find(provided_sig.first) == signature_keys.end())
+         result.insert(provided_sig.first);
 
    return result;
 }
 
-set<public_key_type> signed_transaction::minimize_required_signatures(
-   const chain_id_type& chain_id,
-   const flat_set<public_key_type>& available_keys,
-   const std::function<const authority*(account_id_type)>& get_active,
-   const std::function<const authority*(account_id_type)>& get_owner,
-   bool allow_non_immediate_owner,
-   uint32_t max_recursion
-   ) const
+set<public_key_type> signed_transaction::minimize_required_signatures(const chain_id_type& chain_id,
+                                                                      const flat_set<public_key_type>& available_keys,
+                                                                      const std::function<const authority*(account_id_type)>& get_active,
+                                                                      const std::function<const authority*(account_id_type)>& get_owner,
+                                                                      bool allow_non_immediate_owner,
+                                                                      bool ignore_custom_operation_required_auths,
+                                                                      uint32_t max_recursion) const
 {
-   set< public_key_type > s = get_required_signatures( chain_id, available_keys, get_active, get_owner, max_recursion );
-   flat_set< public_key_type > result( s.begin(), s.end() );
-
-   for( const public_key_type& k : s )
-   {
-      result.erase( k );
-      try
-      {
-         graphene::chain::verify_authority( operations, result, get_active, get_owner, allow_non_immediate_owner, max_recursion );
+   set<public_key_type> s = get_required_signatures(chain_id, available_keys, get_active, get_owner, allow_non_immediate_owner,
```

Comment:
Nice catch. I wonder why this worked before.
And I wonder where `minimize_required_signatures` is currently used, except in the corresponding unit test?

Reply:
AFAIK Steem uses it in cli_wallet: https://github.com/steemit/steem/blob/v0.20.9/libraries/wallet/wallet.cpp#L688, but we don't use it so far.